### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by thomasconner**
#### Description
This PR updates several parts of the SDK to allow further development.

#### Changes
- Export all modules so they are easily extensible by shims.
- Create aliases using babel to prevent long `../../../` expressions when importing modules
- Expose the rack architecture to allow shims to provide custom implmentations for network and cache middleware
- Clean some minor things with the way modules are exported

### CODE REVIEW

File: test/unit/client.test.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Client } from 'src/client';
+import Client from 'src/client';
```
Comment (tejasranade) [inline test/unit/client.test.js]: @thomasconner Does this `import` change impact apps?

File: package.json [hunk 2]
```diff
@@ -33,31 +33,29 @@
...
-    "es6-error": "^3.2.0",
-    "es6-promise": "^4.0.4",
-    "fast-memory-cache": "^2.0.4",
-    "hellojs": "^1.13.1",
-    "local-storage": "^1.4.2",
-    "lodash": "^4.16.0",
-    "loglevel": "^1.4.1",
-    "promise-queue": "^2.2.3",
-    "qs": "^6.2.0",
+    "append-query": "1.1.0",
```
Comment (thomasconner) [inline package.json]: We should be installing specific versions. For now we haven't had an issue but we are risking the chance that things could not work if a new version is released for a lib we are using.

File: test/unit/client.test.js (hunk 1 above)
Comment (thomasconner) [inline test/unit/client.test.js]: No it won't. Apps would access the `Client` on the `Kinvey` namespace.

File: src/request/src/rack.js [hunk 3]
```diff
@@ -79,28 +79,31 @@ export default class Rack extends Middleware {
...
-export class NetworkRack extends Rack {
+class NetworkRack extends Rack {
   constructor(name = 'Network Rack') {
     super(name);
     this.use(new SerializeMiddleware());
     this.use(new HttpMiddleware());
     this.use(new ParseMiddleware());
   }
 }
+const newtworkRack = new NetworkRack();
```
Comment (thomasconner) [inline src/request/src/rack.js]: Fixed

File: src/export.js [hunk 4]
```diff
@@ -1,28 +1,72 @@
...
+  CacheMiddleware,
+  CacheRack,
+  CacheRequest,
+  Client,
   CustomEndpoint,
   DataStore,
   DataStoreType,
-  FileStore as File,
+  DeltaFetchRequest,
   FileStore as Files,
```
Comment (thomasconner) [inline src/export.js]: Yes I have. We have been using `Kinvey.Files` in the docs for sometime now.

File: src/request/src/middleware/src/http.js [hunk 5]
```diff
@@ -1,56 +1,70 @@
...
-          return resolve({
+      httpRequest({
+        method: method,
+        url: url,
+        headers: headers,
+        body: body,
+        followRedirect: followRedirect,
+        timeout: timeout
+      }, (error, response, body) => {
+        if (isDefined(response) === false) {
```
Comment (thomasconner) [inline src/request/src/middleware/src/http.js]: @tejasranade I added some additional unit tests to cover different kinds of requests.

**Comment by thomasconner** [comment]
@tejasranade can you review the changes you requested and see if they satisfy your review?

**Pull request merged**